pkg/wkspace/minimal.go

```diff
@@ -118,12 +119,24 @@ func (m *MinimalWorkspace) BounceHelm(wait bool, skipArgs ...string) error {
 		_, err := instClient.Run(chart, defaultVals)
 		return err
 	}
+
 	client := action.NewUpgrade(m.HelmConfig)
 	client.Namespace = namespace
 	client.SkipCRDs = true
 	client.Timeout = time.Minute * 10
 	client.Wait = wait
 	_, err = client.Run(m.Name, chart, defaultVals)
+	if err != nil {
+		r, errReleases := m.HelmConfig.Releases.Last(m.Name)
+		if errReleases != nil {
+			return errors.Wrap(err, fmt.Sprintf("can't get the last release %v", errReleases))
+		}
+		if !r.Info.Status.IsPending() {
+			return err
+		}
+		rollback := action.NewRollback(m.HelmConfig)
+		return rollback.Run(m.Name)
```

Comment:
double check you don't need to specify exsct version (sometimes there can be a list of failed versions in the history and we'd want to rollback to the last deployed one)

Reply:
@michaeljguarino I've checked it, it finds the previous release and prepares a new release object with the previous release's configuration

Reply:
can you send me a pointer to the code so i can understand?

Reply:
@michaeljguarino 
https://github.com/helm/helm/blob/main/pkg/action/rollback.go#L58
and this part is the most important:
https://github.com/helm/helm/blob/main/pkg/action/rollback.go#L94

Reply:
Yea this does what i thought it might, just rolls back to the immediate previous release, which works for the most part, but it's actually possible for a sequence of previous releases to be broken and you will need to traverse the history for the latest nonbroken one

Reply:
@michaeljguarino PTAL, I search for the last deployed version and set it for the rollback

Reply:
I also have fixed helm diff command